baseplate/thrift/baseplate.thrift

```diff
@@ -127,3 +127,161 @@ struct Request {
+    /** This is appropriate when a client request would cause a conflict with
+    the current state of the server.
+    */
+    CONFLICT = 409,
+    /** This is appropriate when the resource requested was once available but is
+    no longer.
+    */
+    GONE = 410,
+    /** This is appropriate when the server is not in a state required to complete
+    the clients request. For example, deleting an entity that has children which must
+    by deleted first.
+    */
+    PRECONDITION_FAILED = 412,
+    /** This is appropriate when the client sends a request that is larger than
+    the limits set by the server, such as when they try to upload a file that is
+    too big.
+    */
+    PAYLOAD_TOO_LARGE = 413,
+    /** This is appropriate when the server is a teapot rather than a coffee maker.
+    */
+    IM_A_TEAPOT = 418,
+    /** This is appropriate when the request was directed at a server that is not
+    able to produce a response. For example, because of connection reuse.
+    */
+    MISDIRECTED_REQUEST = 421,
+    /** This is appropriate when the request is valid but the server is unable to
+    process the request instructions.
+
+    The request should not be retried without modification.
+    */
+    UNPROCESSSABLE_ENTITY = 422,
+    /** This is appropriate when the client tries to operate on a resource that is
+    locked.
+    */
+    LOCKED = 423,
+    /** This is appropriate when the request failed because it depended on another
+    request and that request failed.
+    */
+    FAILED_DEPENDENCY = 424,
+    /** This is appropriate when the server is concerned that the request may be
+    replayed, resulting in a replay attack.
+    */
+    TOO_EARLY = 425,
+    /** This is appropriate if the server requires a precondition to be specified in
+    order to process a request. For example, a server may require clients to specify
+    a version number for a resource that they are trying to update in order to avoid
+    lost updates. If the client does not specify a version number then the server may
+    respond with `PRECONDITION_REQUIRED`.
+    */
+    PRECONDITION_REQUIRED = 428,
+    /** This is appropriate when the client has been rate limited by the server.
+
+    It may be appropriate for the client to retry the request after some time has
+    passed, it is encouraged to use this along with Retryable to communicate to
+    the client when they are able to retry.
+    */
+    TOO_MANY_REQUESTS = 429,
+    /** This is appropriate when the client sends a header value that is too large.
+    */
+    REQUEST_HEADER_FIELDS_TOO_LARGE = 431,
+    /** This is appropriate when the requested resource is unavailable for
+    legal reasons, such as when the content is censored in a country.
+    */
+    UNAVAILABLE_FOR_LEGAL_REASONS = 451,
+    /** This is appropriate for generic, unhandled server errors.
+    */
+    INTERNAL_SERVER_ERROR = 500,
+    /** This applies when a request is made for an HTTP method that the server
```

Comment:
does it have to be an HTTP method?

Reply:
This one might just not make sense in thrift?